✨(elasticsearch) add Docker image for Elasticsearch 5.6.16

madmatah opened the pull request:
## Purpose

 We need to deploy an Elasticsearch 5.x to use with Ashley forum.
Elasticsearch's offical images in version 5.x cannot be run with an unprivileged user.

## Proposal

- [x] Add a docker image for the latest Elasticsearch 5.x version  (5.6.16)
- [x] Update README
- [x] Remove ES images 0.9.0 and 1.5.2 that cannot be built anymore

madmatah committed:
✨(elasticsearch) add Docker image for Elasticsearch 5.6.16

We need to deploy an Elasticsearch 5.x to use with Ashley forum.
Elasticsearch's offical images in version 5.x cannot be run with an
unprivileged user.

madmatah committed:
📝(README) add missing elasticsearch tags

Some images have been added with no tag documented.

jmaupetit commented on docker/images/elasticsearch/5.6.16/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+# Make sure that the initial directories have the correct permissions
+# for OpenShift. OpenShift will run the elasticsearch with a user wit
+# a random UID and a GID 0.
+USER root
```
Why switching back to `root` user twice?

madmatah replied:
Because I was tired, I guess :man_facepalming: 
Thanks!

madmatah committed:
🔥(elasticsearch) remove elasticsearch 0.90 and 1.5.2

We cannot build docker images for elasticsearch 0.90 and 1.5.2
anymore, because the jessie backports repository has been removed. And
it's a blocking problem for our CircleCI jobs.

These images are still available on DockerHub. Links are in the
README.
